p2panda-rs/src/hash/hash.rs

```diff
@@ -113,17 +116,30 @@ impl Human for Hash {
     }
 }
 
+impl Serialize for Hash {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            self.as_str().serialize(serializer)
+        } else {
+            ByteBuf::from(self.to_bytes()).serialize(serializer)
+        }
+    }
+}
```

Comment:
Here we could maybe also use the utility method or even serde with attribute?

Reply:
You mean the existing util or a new one? The existing one doesn't do what we want, and I thought maybe there wasn't need to create a new util for this one case. Don't feel very strongly about it though.

Reply:
The reason I didn't just make `Hash` contain bytes (which was another valid approach) is that all our APIs around `Hash`, `OperationId`, `DocumentId` etc.. assume strings are the core data type. And I think this is actually nice and fine to keep. Serializing now takes account of the knowledge that this `String` is actually a hex representation of bytes.

Reply:
> The reason I didn't just make `Hash` contain bytes (which was another valid approach) is that all our APIs around `Hash`, `OperationId`, `DocumentId` etc.. assume strings are the core data type. And I think this is actually nice and fine to keep. 
> 
> The change is that Serializing now takes account of the knowledge that this `String` is actually a hex representation of bytes.

Ah, that was my thought, thanks for clarifying. Hmn, I'm fine with leaving it as is for now but wonder if we want to keep it like that. Guess it's performance optimization related and too much guessing at this stage.

Reply:
Yeh, have a look. I think it means a lot of changes to our current APIs, which I thought was worth avoiding. But happy for a second opinion.